src/main/scala/com/softwaremill/react/kafka/KafkaActorSubscriber.scala

```diff
@@ -23,17 +22,12 @@ private[kafka] class KafkaActorSubscriber[T](
       processElement(element.asInstanceOf[T])
     case ActorSubscriberMessage.OnError(ex) =>
       handleError(ex)
-    case ActorSubscriberMessage.OnComplete =>
+    case ActorSubscriberMessage.OnComplete | "Stop" =>
       cleanupResources()
   }
 
   private def processElement(element: T) = {
-    try {
-      producer.send(encoder.toBytes(element), partitionizer(element))
-    }
-    catch {
-      case e: Exception => handleError(e)
-    }
+    producer.send(props.encoder.toBytes(element), props.partitionizer(element))
```

Comment:
`handleError()` in the semantics of reactive streams API is meant to be called only on fatal issues, where you cannot recover and you need to close your resources. It cannot be called twice. This case means that something bad happened when we tried to obtain an element from Kafka which may be recoverable.
I decided to leave error handling to standard mechanisms of Akka - supervision. Throwing an exception should be handled by actor's supervisor. If there's no supervisor, the default Akka's behavior will be writing to logs and restarting the actor (which has no effect on internal Kafka reference) and letting it work further.